bliss/models/location_encoder.py

```diff
@@ -305,87 +303,81 @@ def get_n_source_prior_log_prob(self, detection_rate):
             `n_samples x n_ptiles x max_detections x ...`
-            or `n_ptiles x max_detections x ...` depending on the shape of `tile_n_sources`.
         """
-
         tile_n_sources = tile_n_sources.clamp(max=self.max_detections)
-        if len(tile_n_sources.shape) == 1:
-            tile_n_sources = tile_n_sources.unsqueeze(0)
-            squeeze = True
-        elif len(tile_n_sources.shape) == 2:
-            squeeze = False
-        else:
-            raise ValueError("tile_n_sources must have shape size 1 or 2")
-
-        assert var_params_flat.shape[0] == tile_n_sources.shape[1]
-        # get probability of params except n_sources
-        # e.g. loc_mean: shape = (n_samples x n_ptiles x max_detections x len(x,y))
-        var_params_for_n_sources = self._get_var_params_for_n_sources(
-            var_params_flat, tile_n_sources
-        )
+        n_ptiles = var_params_flat.size(0)
+        vpf_padded = F.pad(var_params_flat, (0, len(self.variational_params), 0, 0))
+
+        var_params_for_n_sources = {}
+        for k, param in self.variational_params.items():
+            indx_mat = getattr(self, k + "_indx")
+            indices = indx_mat[tile_n_sources.transpose(0, 1)].reshape(n_ptiles, -1)
+            var_param = torch.gather(vpf_padded, 1, indices)
+
+            var_params_for_n_sources[k] = rearrange(
+                var_param,
+                "np (ns d pd) -> ns np d pd",
+                np=n_ptiles,
+                ns=tile_n_sources.size(0),
+                d=self.max_detections,
+                pd=param["dim"],
+            )
+
+        # what?!? why is sigmoid(0) = 0?
+        loc_mean_func = lambda x: torch.sigmoid(x) * (x != 0).float()
+        # and why does location mean need to be transformed?
+        # can't we stick with the original logistic-normal parameterization here?
+        var_params_for_n_sources["loc_mean"] = loc_mean_func(var_params_for_n_sources["loc_mean"])
 
-        if squeeze:
-            var_params_for_n_sources = {
-                key: value.squeeze(0) for key, value in var_params_for_n_sources.items()
-            }
         return var_params_for_n_sources
 
     @property
     def variational_params(self):
-        # transform is a function applied directly on NN output.
         return {
-            "loc_mean": {"dim": 2, "transform": _loc_mean_func},
-            "loc_logvar": {"dim": 2, "transform": _identity_func},
-            "log_flux_mean": {"dim": self.n_bands, "transform": _identity_func},
-            "log_flux_logvar": {"dim": self.n_bands, "transform": _identity_func},
+            "loc_mean": {"dim": 2},
+            "loc_logvar": {"dim": 2},
+            "log_flux_mean": {"dim": self.n_bands},
+            "log_flux_logvar": {"dim": self.n_bands},
         }
 
-    # These methods are only used in testing or case studies, Do we need them or can
-    # they be moved to the code that they test? ------------------------
```

Comment:
Not sure what methods this comment referred to. The methods below were in use.

Reply:
I made this comment when I refactored it last but I ended up deleting those methods - it should be removed.